File: ProjectManagement.API/Domain/Entities/Project.cs
namespace ProjectManagement.API.Domain.Entities;

public class Project
{
    public Guid Id { get; set; } = Guid.CreateVersion7();

    public required string Name { get; set; }

    public required string CustomerCompany { get; set; }

    public required string ExecutorCompany { get; set; }

    public DateOnly StartDate { get; set; }

    public DateOnly EndDate { get; set; }

    public int Priority { get; set; }

    public Guid ManagerId { get; set; }

    public Employee Manager { get; set; } = null!;

    public ICollection<Employee> Employees { get; } = [];

    public ICollection<ProjectDocument> Documents { get; } = [];
}

File: Frontend/src/types/project.ts
import type {EmployeeResponse, EmployeeShortResponse} from "./employee.ts";
import type {ProjectDocumentResponse} from "./document.ts";

export interface ProjectResponse {
    id: string;
    name: string;
    customerCompany: string;
    executorCompany: string;
    startDate: string;
    endDate: string;
    priority: number;
    manager: EmployeeShortResponse;
}

export interface ProjectDetailsResponse {
    id: string;
    name: string;
    customerCompany: string;
    executorCompany: string;
    startDate: string;
    endDate: string;
    priority: number;
    manager: EmployeeResponse;
    employees: EmployeeResponse[] | null;
    documents?: ProjectDocumentResponse[];
}

export interface CreateProjectRequest {
    name: string;

    customerCompany: string;

    executorCompany: string;

    startDate: string;

    endDate: string;

    priority: number;

    managerId: string;

    employeeIds?: string[];
}

export interface UpdateProjectRequest {
    name: string;

    customerCompany: string;

    executorCompany: string;

    startDate: string;

    endDate: string;

    priority: number;

    managerId: string;

    employeeIds?: string[];
}

export interface GetProjectsRequest {
    search?: string;
    priority?: number;
    managerId?: string;
    startDateFrom?: string;
    startDateTo?: string;
    sortBy?: string;
    descending?: boolean;
}

